tests/api.c

```diff
@@ -16483,6 +16616,66 @@ static void test_wc_CertPemToDer(void)
 #endif
 }
 
+static void test_wc_PubKeyPemToDer(void)
+{
+#ifdef WOLFSSL_PEM_TO_DER
+#if defined(WOLFSSL_CERT_EXT) || defined(WOLFSSL_PUB_PEM_TO_DER)
+    int ret;
+    const char* key = "./certs/ecc-client-keyPub.pem";
+    byte* cert_buf = NULL;
+    size_t cert_sz = 0, cert_dersz = 0;
+    byte* cert_der = NULL;
+
+    printf(testingFmt, "wc_PubKeyPemToDer()");
+
+
+    ret = wc_PubKeyPemToDer(cert_buf, (int)cert_sz,
+        cert_der, (int)cert_dersz);
+    AssertIntGE(ret, BAD_FUNC_ARG);
+
+    ret = load_file(key, &cert_buf, &cert_sz);
+    if (ret == 0) {
+        cert_dersz = cert_sz; /* DER will be smaller than PEM */
+        cert_der = (byte*)malloc(cert_dersz);
+        if (cert_der) {
+            ret = wc_PubKeyPemToDer(cert_buf, (int)cert_sz,
+                cert_der, (int)cert_dersz);
+            AssertIntGE(ret, 0);
+        }
+    }
+
+    if (cert_der)
+        free(cert_der);
+    if (cert_buf)
+        free(cert_buf);
+#endif
+#endif
+}
+
+static void test_wc_PemPubKeyToDer(void)
+{
+#if defined(WOLFSSL_CERT_EXT) || defined(WOLFSSL_PUB_PEM_TO_DER)
+    int ret;
+    const char* key = "./certs/ecc-client-keyPub.pem";
+    size_t cert_dersz = 1024;
+    byte* cert_der = (byte*)malloc(cert_dersz);
+
+    printf(testingFmt, "wc_PemPubKeyToDer()");
+
+#if 0 /* NULL filename causes valgrind failure */
+    ret = wc_PemPubKeyToDer(NULL, cert_der, (int)cert_dersz);
```

Comment:
Eric, can you fix `wc_PemPubKeyToDer` to check `fileName` for NULL? That function automatically does `XFILE  file    = XFOPEN(fileName, "rb");` without any checks, which should be fixed.

Reply:
Sure. It seemed like that was the intended functionality (i.e. let fopen handle the error). I will fix it up.

Reply:
I made the same fix for `wc_PemCertToDer` as well.